From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: gngpp <gngppz@gmail.com>
Date: Thu, 18 Sep 2026 00:00:00 +0800
Subject: [PATCH 11/11] boringssl: prefix struct tags with out-of-line C++
 destructors

Copyright (c) 2025-2026 0x676e67 <gngppz@gmail.com>

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
---
 include/openssl/base.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/openssl/base.h b/include/openssl/base.h
--- a/include/openssl/base.h
+++ b/include/openssl/base.h
@@ -343,6 +343,11 @@
 typedef struct ssl_method_st SSL_METHOD;
 typedef struct ssl_private_key_method_st SSL_PRIVATE_KEY_METHOD;
 typedef struct ssl_quic_method_st SSL_QUIC_METHOD;
+// This opaque type has out-of-line C++ destructors outside the bssl namespace,
+// so `prefix_symbols.h`, which only renames C symbols, cannot reach them.
+#if defined(BORINGSSL_PREFIX)
+#define ssl_session_st BORINGSSL_ADD_PREFIX(ssl_session_st)
+#endif
 typedef struct ssl_session_st SSL_SESSION;
 typedef struct ssl_st SSL;
 typedef struct ssl_ticket_aead_method_st SSL_TICKET_AEAD_METHOD;
